### PR TITLE
pass storageMarker as argument when SyncAccounts() is called

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**
## Reasoning behind the pull request
- When a missing trie node is found, the subtrie starting from that hash will be synced. This is triggered on `SyncBlock()`. However, the storage should not be marked as synced and active - not the whole trie was synced, but just a missing subtrie.
  
## Proposed changes
- Pass `storageMarker` to `SyncAccounts()` func. This way, a disabled storage marker can be given in case the db should not be marked as synced.

## Testing procedure
- Normal testing procedure with missing trie nodes.

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: state/syncer/userAccountsSyncer.go [hunk 1]
```diff
@@ -104,7 +102,7 @@ func NewUserAccountsSyncer(args ArgsNewUserAccountsSyncer) (*userAccountsSyncer,
 }
 
 // SyncAccounts will launch the syncing method to gather all the data needed for userAccounts - it is a blocking method
-func (u *userAccountsSyncer) SyncAccounts(rootHash []byte) error {
+func (u *userAccountsSyncer) SyncAccounts(rootHash []byte, storageMarker common.StorageMarker) error {
 	u.mutex.Lock()
```
Comment (iulianpascalau) [inline state/syncer/userAccountsSyncer.go:106]: Please add here a test for nil storageMarker argument. The code can panic on L135. You need to add CheckIfNil on the StorageMarker interface. Valid for validatorAccountsSyncer.go L82. 
+ unit tests

Comment (BeniaminDrasovean) [reply state/syncer/userAccountsSyncer.go:106]: Done.

**Review by iulianpascalau — APPROVED** [review]
